assets/javascripts/discourse/components/modal/diff-modal.gjs

```diff
@@ -1,15 +1,76 @@
 import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
 import { action } from "@ember/object";
+import { next } from "@ember/runloop";
+import { inject as service } from "@ember/service";
 import { htmlSafe } from "@ember/template";
+import CookText from "discourse/components/cook-text";
 import DButton from "discourse/components/d-button";
 import DModal from "discourse/components/d-modal";
+import { ajax } from "discourse/lib/ajax";
 import i18n from "discourse-common/helpers/i18n";
 
 export default class ModalDiffModal extends Component {
+  @service currentUser;
+  @tracked loading = false;
+  @tracked diff;
+  suggestion = "";
+
+  PROOFREAD_ID = -303;
+
+  constructor() {
+    super(...arguments);
+    this.diff = this.args.model.diff;
+
+    next(() => {
+      if (this.args.model.selected && this.args.model.toolbarEvent) {
+        this.loadDiff();
+      }
+    });
+  }
+
+  async loadDiff() {
+    this.loading = true;
+
+    const suggestion = await ajax("/discourse-ai/ai-helper/suggest", {
```

Comment:
Should wrap async ajax in try/catch with `popupAjaxError` for failures and reset this.loading in `finally`